velox/common/file/FileSystems.h

```diff
@@ -41,6 +43,11 @@ class FileSystem {
   virtual std::unique_ptr<WriteFile> openFileForWrite(
       std::string_view path) = 0;
 
+  // Deletes the file at 'path'. Throws on error.
+  virtual void remove(std::string_view path) {
```

Comment:
Would it be cleaner to make this method pure virtual and have each implementation decide whether to support it or not?